docs/en/middleware/supplemental.md

```diff
@@ -0,0 +1,521 @@
+-   `i18n.set(localeData?: LocaleData)`
+    -   Allows applications to set a specific locale.
+-   `i18n.get()`
+    -   Returns the currently-set locale, or `undefined` if no theme has been set. Typically used within an application's root widget.
+
+### `icache`
+
+Composes [`cache`](#cache) and [`invalidator`](#invalidator) middleware functionality to provide a cache that supports lazy value resolution and automatic widget invalidation once a value becomes available.
+
+**API:**
+
+```ts
+import icache from '@dojo/framework/core/middleware/icache';
+```
+
+-   `icache.getOrSet<T = any>(key: any, value: any): T | undefined`
+    -   Retrieves the cached value for the given `key`, if one exists, otherwise `value` is set. In both instances, `undefined` is returned if the cached value has not yet been resolved.
+-   `icache.get<T = any>(key: any): T | undefined`
+    -   Retrieves the cached value for the given `key`, or `undefined` if either no value has been set, or if the value is still pending resolution.
+-   `icache.set(key: any, value: any)`
+    -   Sets the provided `value` for the given `key`. If `value` is a function, it will be invoked in order to obtain the actual value to cache. If the function returns a promise, a 'pending' value will be cached until the final value is fully resolved. In all scenarios, once a value is available and has been stored in the cache, the widget will be marked as invalid so it can be re-rendered with the final value available.
+-   `clear()`
+    -   Clears all values currently stored in the widget's local cache.
+
+### `injector`
+
+Allows retrieving injectors from the Dojo registry and assigning invalidation callback functions to then.
+
+**Note:** Injectors and the registry are advanced concepts not typically required when writing Dojo applications. They are mainly used internally by the framework to implement more advanced user-facing functionality such as [Dojo stores](../stores/basic-usage.md).
+
+**API:**
+
+```ts
+import injector from '@dojo/framework/core/middleware/injector';
+```
+
+-   `injector.subscribe(label: RegistryLabel, callback: Function = invalidator)`
+    -   Subscribes the given `callback` invalidation function against the specified registry `label` injector (if one exists). If a `callback` is not specified, the [`invalidator`](#invalidator) middleware is used by default so that the current widget will be marked as invalid and re-rendered when the injector makes its data available.
+-   `injector.get<T>(label: RegistryLabel): T | null`
+    -   Retrieves the current injector associated with the given registry `label`, or `null` if no such injector exists.
+
+### `intersection`
+
+Provides information on whether a node is visible in a particular viewport using the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).
```

Comment:
It might be worth noting that we currently focus on v1 of the API and not v2 ( https://developers.google.com/web/updates/2019/02/intersectionobserver-v2 )

Reply:
Added a note accordingly.